### DISCUSSION TITLE
Nova 4 seems to specify incorrect type hint

### DISCUSSION THREAD

**Discussion by potsky** [laravel/nova-issues#3904] [Q&A] [answered]
- Laravel Version: 8.83.7
- Nova Version: 4.0.5
- PHP Version: 8.0.12
- Database Driver & Version: PGSQL
- Operating System and Version: sail 1.13.9
- Browser type and version: Brave (Chrome) [Version 1.37.109 Chromium: 100.0.4896.60 (Official Build) (arm64)]
- Reproduction Repository: https://github.com/private

### Description:

in `vendor/laravel/nova/src/AuthorizedToSee.php`:

```php
/**
 * Determine if the filter or action should be available for the given request.
 *
 * @param  \Illuminate\Http\Request  $request
 * @return bool
 */
public function authorizedToSee(Request $request)
{
    return $this->seeCallback ? call_user_func($this->seeCallback, $request) : true;
}
```

The request injected in the call back seems to be a `\Illuminate\Http\Request` but when looking in calls up, it is now a `NovaRequest`. So static analysers like PHPStan fail when using something as:

```php
->canSee(function ($request) {
    $query = ($request instanceof ActionRequest) ? $request->findModelQuery($request->resources) : $request->findModelQuery();

    return $query->first()...
}),
```

Indeed, `findModelQuery` does not exist on a `\Illuminate\Http\Request`...

**Comment by crynobone** [comment, marked as answer]
`authorizedToSee` may be called from `Illuminate\Http\Request` (outside of Nova) while most internal Nova requests will always be `NovaRequest`.
